query_execution/QueryContext.hpp

```diff
@@ -184,7 +184,7 @@ class QueryContext {
   /**
    * @brief Release the given AggregationOperationState.
    *
-   * @param id The id of the AggregationOperationState to destroy.
+   * @param id The id of the AggregationOperationState to release.
```

Comment:
@hbdeshmukh Since we changed the semantics of `FinalizeAggregationOperator`, when constructing `FinalizeAggregationWorkOrder`, [`WorkOrderFactory`](https://github.com/apache/incubator-quickstep/pull/96/files#diff-2c12385b34f03b5eb61a3f5df00505eeR157) should also use the get version instead. And it fixes the distributed unit tests. Otherwise, it would also cause memory leaks for the distributed version.

@pateljm The test failures are related to the missed changes in new feature for the distributed version, not my existing code. The main components in the distributed query execution engine was code reviewed, although not the test drivers (but the most driver code is forked from the existing codebase).